apps/web/src/App/Pages/Auth/SignUp/index.tsx

```diff
@@ -1,5 +1,145 @@
+
+  return (
+    <Container>
+      <Card>
+        <CardHeader>
+          <Heading size='xl'>{t(d.registration)}</Heading>
+        </CardHeader>
+        <CardBody>
+          <form onSubmit={handleSubmit(registration)} noValidate>
+            <Stack spacing={6}>
+              <FormControl isInvalid={!!errors.name || !!errors.root?.serverError}>
+                <FormLabel>{t(d.NameUser)}</FormLabel>
+                <Input
+                  id='name'
+                  type='name'
+                  placeholder={t(d.Name)}
+                  {...register('name', {
+                    required: t(d.NameIsRequired),
+                    minLength: MinimalLength(t, d),
+                    maxLength: MaximumLength(t, d),
+                  })}
+                />
+                <FormErrorMessage>{errors.name && errors.name.message}</FormErrorMessage>
+              </FormControl>
+              <FormControl isInvalid={!!errors.email || !!errors.root?.serverError}>
+                <FormLabel>{t(d.EmailAddress)}</FormLabel>
+                <Input
+                  id='email'
+                  type='email'
+                  placeholder={t(d.Email)}
+                  {...register('email', {
+                    required: t(d.EmailIsRequired),
+                    minLength: MinimalLength(t, d),
+                    maxLength: MaximumLength(t, d),
+                    pattern: {
+                      value: RegularEmailValidation,
+                      message: t(d.InvalidEmailAddress),
+                    },
+                  })}
+                />
+                <FormErrorMessage>{errors.email && errors.email.message}</FormErrorMessage>
+              </FormControl>
+              <FormControl isInvalid={!!errors.password || !!errors.root?.serverError}>
+                <FormLabel>{t(d.Password)}</FormLabel>
+                <Input
+                  id='password'
+                  type='password'
+                  placeholder={t(d.Password)}
+                  {...register('password', {
+                    required: t(d.PasswordIsRequired),
+                    minLength: MinimalLength(t, d),
+                    maxLength: MaximumLength(t, d),
+                    onChange: (e: SyntheticEvent<HTMLInputElement>) => setPassword(e.currentTarget.value),
+                  })}
+                />
+                <FormErrorMessage>{errors.password && errors.password.message}</FormErrorMessage>
+              </FormControl>
+              <FormControl isInvalid={!!errors.repeatPassword || !!errors.root?.serverError}>
+                <Input
+                  id='repeatPassword'
+                  type='password'
+                  placeholder={t(d.RepeatPassword)}
+                  {...register('repeatPassword', {
+                    required: t(d.PasswordIsRequired),
+                    minLength: MinimalLength(t, d),
+                    maxLength: MaximumLength(t, d),
+                    validate: (val) => {
```

Comment:
Давай вот так сократим:
```suggestion
                    (val) => password === val || t(d.PasswordsNotMatch)
```